src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2065 @@
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 5][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 5][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 5][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 6][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 6][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 6][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 7][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 7][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 7][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 8][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 72);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 8][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 72);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 8][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 72);
+        rpp_hip_adjust_range(dstPtr, &sum_f24);
+        rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &sum_f24);
+    }
+}
+
+__device__ float gaussian(int x, int y, float mulFactor)
+{
+    float expFactor = - ((x * x) + (y * y)) * mulFactor;
+    expFactor = expf(expFactor);
+    float res  = (expFactor * mulFactor) / PI;
+    return res;
+}
+
+__global__ void set_filter_values(float *filterTensor, float *stdDevTensor, int kernelSize, int N, int batchSize)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x);
+    if(id_x >= batchSize)
+        return;
+
+    float *filter = &filterTensor[id_x * N];
+    float stdDev = stdDevTensor[id_x];
+    int cnt = 0;
+    float mulFactor = 1 / (2 * stdDev * stdDev);
+    float kernelSum = 0.0f;
+    for(int i = 0; i < kernelSize; i++)
+    {
+        for(int j = 0; j < kernelSize; j++)
+        {
+            filter[cnt] = gaussian(i, j, mulFactor);
+            kernelSum += filter[cnt];
+            cnt++;
+        }
+    }
+    kernelSum = (kernelSum == 0.0f) ? 1.0f : (1.0f / kernelSum);
+
+    // Multiply kernel values by (1 / kernelSum)
+    cnt = 0;
+    for(int i = 0; i < kernelSize; i++)
+    {
+        for(int j = 0; j < kernelSize; j++)
+        {
+            filter[cnt] *= kernelSum;
```

Comment:
We shouldn't independently write 81 floats to global. Accumulate all row-replications, kernel computations in a d_float81 and issue a write call as *(d_float81 *)filter = filter_f81;

Reply:
create_gaussian_kernel is a common kernel to all kernel sizes

if we need to update in temporary we need to add a branch condition based on kernel size inside the hip kernel
I am not sure even after making this local variable use because the scope will be only inside if condition

it will be something like below
`if (kernelSize == 3)
   d_float9 filter_f9;
else if (kernelSize == 5)
  d_float25 filter_f25;
else if (kernelSize == 5)
  d_float49 filter_f49;
else if (kernelSize == 9)
  d_float81 filter_f81;`

This variable cannot be used outside the if conditions, we have 2 options
1. we need to replicate the same code for all kernel sizes inside the if conditions
2. we can have seperate kernels for filter generation based on filter generation. Please let me know your thoughts on this @r-abishek

Reply:
Done